src/include/miopen/conv/heuristic_model/tuning_heuristic.hpp

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+#define GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+
+#if MIOPEN_ENABLE_AI_HEUR
+#include <fdeep/fdeep.hpp>
+#include <fdeep/tensor.hpp>
+#include <fdeep/tensor_shape.hpp>
+#include <miopen/conv/context.hpp>
+#include <miopen/solver.hpp>
+#include <unordered_map>
+#include <queue>
+#include <typeinfo>
+#include <string>
+
+namespace miopen {
+namespace ai {
+
+inline nlohmann::json get_metadata(const std::string& arch, const std::string& solver)
+{
+    std::string file_path = GetSystemDbPath() + "/" + arch + "_" + solver + "_metadata.model";
+    return nlohmann::json::parse(std::ifstream(file_path));
+}
+
+inline fdeep::model
+get_model(const std::string& arch, const std::string& solver, const std::string& model_type)
+{
+    std::string file_path =
+        GetSystemDbPath() + "/" + arch + "_" + solver + "_" + model_type + ".model";
+    return fdeep::load_model(file_path, true, fdeep::dev_null_logger);
+}
+
+inline bool model_set_params(const fdeep::model& encoder,
+                             const fdeep::model& decoder,
+                             const nlohmann::json& metadata,
+                             solver::PerformanceConfigConvAsm1x1U& config,
+                             const ProblemDescription& problem,
+                             std::vector<float>& features)
```

Comment:
1. Please use `CamelCase` for all function names (like in other places).
2. From the function name it is not clear what kind of parameters we are setting.
3. Do you need this function to be inline? If not please put only declaration in the header.

Reply:
This function needs to be inline, as well as GetModel(). This is necessary since we would need to introduce a pimpl coding pattern to the fdeep::model object to not get duplicate definition errors. This will be done in future PRs.

Reply:
@Dmantri98 I believe that proposed future changes should not be allowed to spoil today's design.

[Notice]
> ...we would need to introduce a pimpl coding pattern to the fdeep::model object to not get duplicate definition errors

As pimpl idiom introduces run-time overhead (indirection plus heap usage) and increases maintenance costs I will ask for more detailed justification (when the time comes, not now).